hack/ci/run-argocd-e2e-tests.sh

```diff
@@ -0,0 +1,313 @@
+    echodate AWS credentials found! Proceeding.
+  elif [[ -n "${AWS_PROFILE-}" ]]; then
+    echodate AWS profile variable found! Proceeding.
+  else
+    echodate No AWS credentials configured. You must export either combination of AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY OR AWS_PROFILE env variable. Exiting the script.
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v git)" ]; then
+    echodate 'Error: git is not installed.' >&2
+    exit 1
+  fi
+
+  mkdir -p ${KUBEONE_INSTALL_DIR}
+  curl -sLO "https://github.com/kubermatic/kubeone/releases/download/v${K1_VERSION}/kubeone_${K1_VERSION}_linux_amd64.zip" &&
+    unzip -qq kubeone_${K1_VERSION}_linux_amd64.zip -d kubeone_${K1_VERSION}_linux_amd64 &&
+    mv kubeone_${K1_VERSION}_linux_amd64/kubeone ${KUBEONE_INSTALL_DIR} && rm -rf kubeone_${K1_VERSION}_linux_amd64 kubeone_${K1_VERSION}_linux_amd64.zip
+
+  if ! [ -x ${KUBEONE_INSTALL_DIR}/kubeone ]; then
+    echodate 'Error: kubeone is not installed.' >&2
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v helm)" ]; then
+    echodate 'Error: helm is not installed.' >&2
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v sed)" ]; then
+    echodate 'Error: sed is not installed.' >&2
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v tofu)" ]; then
+    echodate 'Error: tofu is not installed.' >&2
+    exit 1
+  fi
+
+  cd /tmp
+  curl -sL https://github.com/kyverno/chainsaw/releases/download/v${CHAINSAW_VERSION}/chainsaw_linux_amd64.tar.gz | tar -xz
+  mv chainsaw /usr/local/bin
+  cd -
+
+  if ! [ -x "$(command -v chainsaw)" ]; then
+    echodate 'Error: chainsaw testing tool is not installed.' >&2
+    exit 1
+  fi
+
+  # TODO: Review if we really need to save things once CI starts to work perfectly.
+  # download and setup AWS CLI
+  curl -s "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+  unzip -q awscliv2.zip
+  ./aws/install
+  rm -rf awscliv2.zip ./aws
+}
+
+function restoreSshKey() {
+  echodate "Downloading SSH key pair from s3"
+  aws s3 cp s3://cluster-backup-e2e/kkp-argocd-test/ssh-keys/id_rsa ~/.ssh/id_rsa
+  aws s3 cp s3://cluster-backup-e2e/kkp-argocd-test/ssh-keys/id_rsa.pub ~/.ssh/id_rsa.pub
+  chmod 400 ~/.ssh/id_rsa
+  eval $(ssh-agent -s) && ssh-add ~/.ssh/id_rsa
+}
+
+checkoutTestRepo() {
+  echodate "Cloning the argocd gitops Git Repo"
+  ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+  git clone git@github.com:kubermatic-labs/kkp-using-argocd.git
+}
+
+createSeedClusters() {
+  echodate creating Seed Clusters
+  # export TF_LOG=DEBUG
+  cd kubeone-install/${MASTER} && tofu init && tofu apply -auto-approve &&
+    tofu output -json > tf.json && ../../../${KUBEONE_INSTALL_DIR}/kubeone apply -t ./tf.json -m kubeone.yaml --auto-approve
+  if [ $? -ne 0 ]; then
+    echodate kubeone master cluster installation failed.
+    exit 2
+  fi
+  cd ../..
+  aws s3 cp ${MASTER_KUBECONFIG} s3://cluster-backup-e2e/kkp-argocd-test/kubeconfig/
```

Comment:
Will this conflict if multiple ArgoCD e2e jobs run at the same time?

Reply:
It will not conflict... because - we retrieve the terraform state from s3 as well as AWS key and so same kubeone clusters will be taking `kubeone apply` command so same kubconfig will be generated and saved.